### ISSUE TITLE
[gRPC] DNS resolution failed for service

### ISSUE THREAD

**Issue by chenaoxd** [telepresenceio/telepresence#1871]
GRPC will report DNS resolution failed when with `telepresence connect`ed on macos.

The error message is listed below:
```
  File "/Users/foo/bar/venv/lib/python3.7/site-packages/etcd3/client.py", line 1186, in client
    grpc_options=grpc_options)
  File "/Users/foo/bar/venv/lib/python3.7/site-packages/etcd3/client.py", line 148, in __init__
    resp = self.auth_stub.Authenticate(auth_request, self.timeout)
  File "/Users/foo/bar/venv/lib/python3.7/site-packages/grpc/_channel.py", line 923, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/Users/foo/bar/venv/lib/python3.7/site-packages/grpc/_channel.py", line 826, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "DNS resolution failed for service: http://foo.somehost.com"
        debug_error_string = "{"created":"@1626334038.200779000","description":"Resolver transient failure","file":"src/core/ext/filters/client_channel/resolving_lb_policy.cc","file_line":220,"referenced_errors":[{"created":"@1626334038.200765000","description":"DNS resolution failed for service: http://foo.somehost.com","file":"src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc","file_line":378,"grpc_status":14,"referenced_errors":[{"created":"@1626334038.200675000","description":"C-ares status is not ARES_SUCCESS qtype=A name=http is_balancer=0: Domain name not found","file":"src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc","file_line":287,"referenced_errors":[{"created":"@1626334038.199993000","description":"C-ares status is not ARES_SUCCESS qtype=AAAA name=http is_balancer=0: Domain name not found","file":"src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc","file_line":287}]}]}]}"
```

**Comment by chenaoxd** [comment]
After a period of trying, I found it's related to the gRPC C-Ares issues. Adding the `GRPC_DNS_RESOLVER=native` environment variable will resolve my issue. And here's the related issue: https://github.com/grpc/grpc/issues/19954.

Thanks and telepresence is a really good tool!

**Comment by astro-stan** [comment]
I know this has been closed for a long time and I am using GRPC for something totally different but I had the same issue with the python GRPC client. It was working fine for weeks until one day it just stopped with the above error. The environment variable above made no difference.


After hours of fiddling with it I found the error.  Turns out I was accidentally using a target URL with a specified scheme ("https://"). Apparently the python client **must be called without one**:
Example:
```
grpc.insecure_channel("mydomain.com:50051")
```
